mlir/lib/Target/LLVMIR/ModuleTranslation.cpp

```diff
@@ -2332,6 +2332,10 @@ mlir::translateModuleToLLVMIR(Operation *module, llvm::LLVMContext &llvmContext,
   // beforehand.
   translator.debugTranslation->addModuleFlagsIfNotPresent();
 
+  // Call the OpenMP IR Builder callbacks prior to verifying the module
+  if (auto *ompBuilder = translator.getOpenMPBuilder())
+    ompBuilder->finalize();
+
   if (!disableVerification &&
       llvm::verifyModule(*translator.llvmModule, &llvm::errs()))
     return nullptr;
```

Comment:
I think we can move module verification to `~ModuleTranslation` instead. This way we both make sure that verification happens after finalization and we do not need to add the `isFinalized` bool.

Reply:
We could do this, but then we'd have to change the signature of `mlir::translateModuleToLLVMIR` to remove the `disableVerification` option as it would no longer be possible to run the translation without verification, because verification would always happen in the destructor. Is this desirable?

Reply:
I think leaving it here makes more sense indeed. Thanks for looking into it.